src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -3093,16 +3209,113 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 
 		return min;
 	}
-	case SN_Conjugate:
-	case SN_Distance:
-	case SN_DistanceSquared:
+	case SN_Distance: 
+	case SN_DistanceSquared: {
+#if defined(TARGET_ARM64)
+		MonoInst *diffs = emit_simd_ins_for_sig (cfg, klass, OP_XBINOP, OP_FSUB, MONO_TYPE_R4, fsig, args);
+		MonoInst *diffs_squared = emit_simd_ins (cfg, klass, OP_XBINOP, diffs->dreg, diffs->dreg);
+		diffs_squared->inst_c0 = OP_FMUL;
+		diffs_squared->inst_c1 = MONO_TYPE_R4;
+
+		switch (id) {
+		case SN_Distance:
+			return emit_sum_sqrt_vector_2_3_4 (cfg, klass, diffs_squared);
+		case SN_DistanceSquared:
+			return emit_sum_vector (cfg, fsig->params [0], MONO_TYPE_R4, diffs_squared);
+		default:
+			g_assert_not_reached ();
+		}
+#endif
+	}
+	break;
 	case SN_Length:
-	case SN_LengthSquared:
-	case SN_Lerp:
+	case SN_LengthSquared: {
+#if defined (TARGET_ARM64)
+		int src1 = load_simd_vreg (cfg, cmethod, args [0], NULL);
+
+		MonoInst *vec_squared = emit_simd_ins (cfg, klass, OP_XBINOP, src1, src1);
+		vec_squared->inst_c0 = OP_FMUL;
+		vec_squared->inst_c1 = MONO_TYPE_R4;
+
+		switch (id) {
```

Comment:
Same as for `_Distance(Squared)`,  we could use dot product code here and optionally use sqrt depending on the method called.